overlord/devicestate/firstboot_test.go

```diff
@@ -171,7 +179,155 @@ func (s *FirstBootTestSuite) TestPopulateFromSeedErrorsOnState(c *C) {
 	c.Assert(err, ErrorMatches, "cannot populate state: already seeded")
 }
 
+func (s *FirstBootTestSuite) coreSnaps(c *C, withConfigure bool) (seedFragment string) {
+	files := [][]string{}
+	if withConfigure {
+		files = [][]string{{"meta/hooks/configure", ""}}
+	}
+	// put core snap into the SnapBlobDir
+	snapYaml := `name: core
+version: 1.0
+type: os`
+	mockSnapFile := snaptest.MakeTestSnapWithFiles(c, snapYaml, files)
+	targetCore := filepath.Join(dirs.SnapSeedDir, "snaps", filepath.Base(mockSnapFile))
+	err := os.Rename(mockSnapFile, targetCore)
+	c.Assert(err, IsNil)
+
+	snapDeclCore, err := s.storeSigning.Sign(asserts.SnapDeclarationType, map[string]interface{}{
+		"series":       "16",
+		"snap-id":      "core-snap-id",
+		"publisher-id": "canonical",
+		"snap-name":    "core",
+		"timestamp":    time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+
+	sha3_384, size, err := asserts.SnapFileSHA3_384(targetCore)
+	c.Assert(err, IsNil)
+
+	snapRevCore, err := s.storeSigning.Sign(asserts.SnapRevisionType, map[string]interface{}{
+		"snap-sha3-384": sha3_384,
+		"snap-size":     fmt.Sprintf("%d", size),
+		"snap-id":       "core-snap-id",
+		"developer-id":  "canonical",
+		"snap-revision": "1",
+		"timestamp":     time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+
+	writeAssertionsToFile("core.asserts", []asserts.Assertion{snapRevCore, snapDeclCore})
+
+	// put kernel snap into the SnapBlobDir
+	snapYaml = `name: pc-kernel
+version: 1.0
+type: kernel`
+	mockSnapFile = snaptest.MakeTestSnapWithFiles(c, snapYaml, files)
+	targetKernel := filepath.Join(dirs.SnapSeedDir, "snaps", filepath.Base(mockSnapFile))
+	err = os.Rename(mockSnapFile, targetKernel)
+	c.Assert(err, IsNil)
+
+	snapDeclKernel, err := s.storeSigning.Sign(asserts.SnapDeclarationType, map[string]interface{}{
+		"series":       "16",
+		"snap-id":      "pc-kernel-snap-id",
+		"publisher-id": "canonical",
+		"snap-name":    "pc-kernel",
+		"timestamp":    time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+
+	sha3_384, size, err = asserts.SnapFileSHA3_384(targetKernel)
+	c.Assert(err, IsNil)
+
+	snapRevKernel, err := s.storeSigning.Sign(asserts.SnapRevisionType, map[string]interface{}{
+		"snap-sha3-384": sha3_384,
+		"snap-size":     fmt.Sprintf("%d", size),
+		"snap-id":       "pc-kernel-snap-id",
+		"developer-id":  "canonical",
+		"snap-revision": "1",
+		"timestamp":     time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+
+	writeAssertionsToFile("kernel.asserts", []asserts.Assertion{snapRevKernel, snapDeclKernel})
+
+	gadgetYaml := `
+volumes:
+    volume-id:
+        bootloader: grub
+`
+	if withConfigure {
+		gadgetYaml += `
+defaults:
+    foo-snap-id:
+       foo-cfg: foo.
+    core-snap-id:
+       core-cfg: core.
+    pc-kernel-snap-id:
+       pc-kernel-cfg: pc-kernel.
+    pc-snap-id:
+       pc-cfg: pc.
+`
+	}
+
+	// put gadget snap into the SnapBlobDir
```

Comment:
I did that, but then the natural thing was to refactor other tests too, so bigger diff, sorry